### PR TITLE
Add recipes for indi and pyindiclient

### PR DESCRIPTION
**Pull request by joseph-long**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/indi/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - cfitsio
+    - gsl
+    - zlib
+
+test:
+  commands:
+    - test -x $PREFIX/bin/indiserver  # [unix]
+    - test -d $PREFIX/include/libindi/  # [unix]
+    - test -f $PREFIX/lib64/libindiclient.a  # [unix and linux64]
+    - test -f $PREFIX/lib/libindiclient.a  # [unix and not linux64]
```
Comment (scopatz) [inline recipes/indi/meta.yaml]: You should be building shared libraried if you can

File: recipes/indi/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/indilib/indi/archive/v{{ version }}.tar.gz
+  sha256: 82004db3d1e45f8852c1b871fdb925a9ce0afda4f8f92de0a197b9b868933abf
+
+build:
+  number: 0
+  skip: True  # [win]
```
Comment (scopatz) [inline recipes/indi/meta.yaml]: This should probably have a `run_exports` section: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements

File: recipes/pyindiclient/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  skip: True  # [win]
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    # If your project compiles code (such as a C extension) then add the required compilers as separate entries here.
+    # Compilers are named 'c', 'cxx' and 'fortran'.
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - swig
```
Comment (scopatz) [inline recipes/pyindiclient/meta.yaml]: swig is probably a host dep, like cython

File: recipes/indi/meta.yaml (hunk 1 above)
Comment (joseph-long) [reply recipes/indi/meta.yaml]: I looked through their CMakeLists and don't think this lib can be built shared without changes upstream

File: recipes/indi/meta.yaml (hunk 2 above)
Comment (joseph-long) [reply recipes/indi/meta.yaml]: What would I put in it? I read the linked doc but it sounds like listing, e.g., cfitsio as a "host" dependency includes its run_exports implicitly

File: recipes/pyindiclient/meta.yaml (hunk 3 above)
Comment (joseph-long) [reply recipes/pyindiclient/meta.yaml]: changed and pushed

File: recipes/indi/meta.yaml (hunk 1 above)
Comment (scopatz) [reply recipes/indi/meta.yaml]: I think if you set this variable at the command line, you'll get them: https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html

File: recipes/indi/meta.yaml (hunk 2 above)
Comment (scopatz) [reply recipes/indi/meta.yaml]: So this affects your downstream packages, so they don't have to explicitly have to have host & run deps, but can have the host deps propogate

**Pull request closed without merging**